src/awkward/_v2/_connect/cuda/cuda_kernels/awkward_BitMaskedArray_to_ByteMaskedArray.cu

```diff
@@ -0,0 +1,50 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/main/LICENSE
+
+__global__ void
+cuda_BitMaskedArray_to_ByteMaskedArray(int8_t* tobytemask,
+  const uint8_t* frombitmask,
+  int64_t bitmasklength,
+  bool validwhen,
+  bool lsb_order,
+  int64_t invocation_index,
+  int64_t* err_code) {
+  int64_t thread_id = blockIdx.x * blockDim.x + threadIdx.x;
+  if (thread_id < bitmasklength && err_code[0] == MAX_NUMPY_INT) {
```

Comment:
If it's more clear or easier to automate, this could become two separate `if` statements.

Every kernel will need to check `err_code[0] == NO_ERROR`. That becomes something you'd want to copy-paste without thinking.

The `if (thread_id < bitmasklength)` is something you have to think about when writing every kernel: which function argument corresponds to the array length you don't want to overstep can have a different name in every kernel.

I'm 99% sure that two nested `if` statements either generates the same code as one `if` with an `&&` or is otherwise just as fast.